server/src/main/java/io/crate/execution/dml/upsert/InsertSourceGen.java

```diff
@@ -57,11 +57,15 @@ static InsertSourceGen of(TransactionContext txnCtx,
                               NodeContext nodeCtx,
                               DocTableInfo table,
                               String indexName,
-                              GeneratedColumns.Validation validation,
+                              boolean validate,
                               List<Reference> targets) {
         if (targets.size() == 1 && targets.get(0).column().equals(DocSysColumns.RAW)) {
-            return new FromRawInsertSource(table, txnCtx, nodeCtx, indexName);
+            if (!validate && table.generatedColumns().isEmpty() && table.defaultExpressionColumns().isEmpty()) {
```

Comment:
noting that as shown here, skipping validation takes effect only if the table does not have generated/default columns, which has been the original behaviour.